cli/src/trusted_operation.rs

```diff
@@ -227,6 +223,79 @@ fn send_direct_request(
 	}
 }
 
+/// sends a rpc watch request to the worker api server
+pub fn get_json_request(
+	trusted_args: &TrustedArgs,
+	operation_call: &TrustedOperation,
+	shielding_pubkey: sgx_crypto_helper::rsa3072::Rsa3072PubKey,
+) -> String {
+	let operation_call_encrypted = shielding_pubkey.encrypt(&operation_call.encode()).unwrap();
+	let shard = read_shard(trusted_args).unwrap();
+
+	// compose jsonrpc call
+	let request = Request { shard, cyphertext: operation_call_encrypted };
+	RpcRequest::compose_jsonrpc_call(
+		"author_submitAndWatchExtrinsic".to_string(),
+		vec![request.to_hex()],
+	)
+	.unwrap()
+}
+
+pub fn wait_until(
+	receiver: &Receiver<String>,
+	until: fn(TrustedOperationStatus) -> bool,
```

Comment:
I'd suggest changing this parameter to `impl Fn(TrustedOperationStatus) -> bool`. That makes it more flexibel, also allowing closure to be passed in e.g.